include/taskqueue.h

```diff
@@ -0,0 +1,22 @@
+#ifndef __SYS_TASKQUEUE_H__
+#define __SYS_TASKQUEUE_H__
+
+#include <queue.h>
+
+typedef struct task {
+  STAILQ_ENTRY(task) tq_link;
+  void *arg;            /* function argument */
+  void (* func)(void*); /* function pointer */
+} task_t;
+
+typedef struct tq_head taskqueue_t;
```

Comment:
I had a look at NetBSD `workqueue` implementation. You've completely forgotten about synchronization issues...

```C
typedef struct taskqueue {
  mtx_t tq_mutex;
  condvar_t tq_nonempty; /* worker waits on this cv for tq_list to become non empty */
  STAILQ_HEAD(, task) tq_list;
  thread_t *tq_worker;
};
```